pkg/netauth/subtle/util.go

```diff
@@ -0,0 +1,32 @@
+package subtle
+
+import (
+	"bytes"
+	"errors"
+
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	// ErrNoMatchingKeys is returned when no matching keys are found
+	ErrNoMatchingKeys = errors.New("no matching keys found")
+)
+
+// CompareSSHKeys checks if the given SSH public key is in the list of SSH public keys
+func CompareSSHKeys(keys []string, test string) error {
+	pubkey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(test))
+	if err != nil {
+		return err
+	}
+
+	for _, k := range keys {
+		key, _, _, _, err := ssh.ParseAuthorizedKey([]byte(k))
+		if err != nil {
+			continue
+		}
+		if bytes.Equal(key.Marshal(), pubkey.Marshal()) {
```

Comment:
https://pkg.go.dev/github.com/gliderlabs/ssh#KeysEqual does this in a timing safe way.  Just to be safe lets do a timing safe compare here.  I don't see any realistic way to exploit this, but the time cost for doing this timing safe isn't too expensive.